src/common.ts

```diff
@@ -54,4 +55,57 @@ export class Common {
     logger.debug(`loginUrl: ${loginUrl}`);
     return loginUrl;
   }
+
+  public static async identifyPossibleScratchOrgs(fields: AuthFields, orgAuthInfo: AuthInfo): Promise<void> {
+    const logger = await Logger.child('Common', { tag: 'identifyPossibleScratchOrgs' });
+
+    // return if we already know the hub or we know it is a devhub or prod-like
+    if (fields.isDevHub || fields.devHubUsername || fields.loginUrl === 'https://login.salesforce.com') return;
```

Comment:
@mshanemc the prod like check is 100% accurate as a means to exclude an org as a scratch org. We do have the ability to create scratch orgs on production instances through a dev hub org perm and specifying the "instance" property in scratch org def. Not common, but it is an edge case.